src/analysis/inner_product_space/adjoint.lean

```diff
@@ -286,6 +313,24 @@ instance : star_module 𝕜 (E →ₗ[𝕜] E) := ⟨linear_equiv.map_smulₛₗ
 
 lemma star_eq_adjoint (A : E →ₗ[𝕜] E) : star A = A.adjoint := rfl
 
+section unitary
+
+variables {U : (E →ₗ[𝕜] E)}
+
+lemma norm_map_of_unitary (hU : U ∈ unitary (E →ₗ[𝕜] E)) (x : E) : ∥U x∥ = ∥x∥ :=
+  by rw [←sq_eq_sq (norm_nonneg (U x)) (norm_nonneg x), norm_sq_eq_inner,
+    ←adjoint_inner_left, ←mul_apply, ←star_eq_adjoint, unitary.star_mul_self_of_mem hU, one_apply,
+    inner_self_eq_norm_sq]
```

Comment:
Is this proof identical to the one above? If so, the one above should follow immediately from this one.